src/main/java/seedu/address/model/ModelManager.java

```
@@ -29,28 +32,75 @@
-
+        this.queueManager = new QueueManager(queueManager);
         this.addressBook = new AddressBook(addressBook);
         this.appointmentBook = new AppointmentBook(patientSchedule);
         this.userPrefs = new UserPrefs(userPrefs);
-        this.filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
-        this.filteredEvents = new FilteredList<>(this.appointmentBook.getEventList());
+        filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
+        filteredRooms = new FilteredList<>(this.queueManager.getRoomList());
+        filteredEvents = new FilteredList<>(this.appointmentBook.getEventList());
+        filteredReferenceIds = new FilteredList<>(this.queueManager.getReferenceIdList());
     }
 
     public ModelManager() {
-        this(new UserPrefs(), new AddressBook(), new AppointmentBook());
+        this(new AddressBook(), new UserPrefs(), new QueueManager(), new AppointmentBook());
+    }
+
+    //=========== QueueManager ==================================================================================
+
+    @Override
+    public QueueManager getQueueManager() {
+        return queueManager;
+    }
+
+    @Override
+    public void removePatient(ReferenceId target) {
+        queueManager.removePatient(target);
+    }
+
+    @Override
+    public void removePatient(int index) {
+        queueManager.removePatient(index);
+    }
+
+    @Override
+    public void addPatient(ReferenceId id) {
+        queueManager.addPatient(id);
+        updateFilteredReferenceIdList(PREDICATE_SHOW_ALL_ID);
     }
 
+    @Override
+    public void next(int index) {
+        queueManager.serveNext(index);
+    }
+
+    @Override
+    public boolean hasId(ReferenceId id) {
+        requireNonNull(id);
+        return queueManager.hasId(id);
+    }
+
+    @Override
+    public void addRoom(ReferenceId id) {
+        queueManager.addRoom(id);
```

Could you throw an UnsupportedOperationException for now? As you have yet to implement the underlying logic.